packages/design-system/src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -29,9 +37,140 @@ describe('Dropdown', () => {
+    makeDropdown({ defaultValue: '2' }, 4);
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveValue('2');
+  });
+
+  it('applies additional classNames to select element', () => {
+    makeDropdown({ fieldClassName: 'foo' });
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveClass('foo');
+    // Make sure we're not replacing the other class names
+    expect(select).toHaveClass('ds-c-field');
+  });
+
+  it('adds small size classes to select element', () => {
+    makeDropdown({ size: 'small' });
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveClass('ds-c-field--small');
+  });
+
+  it('adds medium size classes to select element', () => {
+    makeDropdown({ size: 'medium' });
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveClass('ds-c-field--medium');
+  });
+
+  it('is inversed', () => {
+    makeDropdown({ inversed: true });
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveClass('ds-c-field--inverse');
+  });
+
+  it('has error', () => {
+    const { container } = makeDropdown({ errorMessage: 'Error' });
+
+    const wrapper = container.querySelector('div');
+    expect(wrapper).toHaveAttribute('aria-invalid', 'true');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('handles bottom placed error', () => {
+    const { container } = makeDropdown({
+      errorMessage: 'Error',
+      errorPlacement: 'bottom',
+      errorId: '1_error',
+    });
+
+    const wrapper = container.querySelector('div');
+    const select = screen.getByRole('combobox');
+    expect(wrapper).toHaveAttribute('aria-invalid', 'true');
+    expect(select).toHaveAttribute('aria-describedby', '1_error');
+    expect(select).toHaveClass('ds-c-field--error');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('is disabled', () => {
+    makeDropdown({ disabled: true });
+
+    const select = screen.getByRole('combobox');
+    expect(select).toHaveAttribute('disabled');
+  });
+
+  it('is not disabled', () => {
+    makeDropdown();
+
+    const select = screen.getByRole('combobox');
+    expect(select).not.toHaveAttribute('disabled');
+  });
+
+  it('calls the onChange handler', () => {
+    makeDropdown({ defaultValue: '1' }, 10);
+    const select = screen.getByRole('combobox');
+
+    userEvent.selectOptions(select, screen.getByRole('option', { name: '2' }));
+    expect(defaultProps.onBlur).not.toHaveBeenCalled();
+    expect(defaultProps.onChange).toHaveBeenCalled();
+  });
+
+  it('calls the onBlur handler', () => {
+    makeDropdown({ defaultValue: '1' }, 10);
+    const selectLabel = screen.getByLabelText('Select an option');
+    const select = screen.getByRole('combobox');
+
+    userEvent.click(selectLabel);
+    expect(select).toHaveFocus();
+    userEvent.tab();
+    expect(defaultProps.onBlur).toHaveBeenCalled();
+    // TODO: reimpliment this to check for no onChange in this test
```

Comment:
Interesting problem I ran into when moving this test to Dropdown, something about how RTL/Jest handles functional components? Even if I move the `onChange` test before all the interaction it fires an `onChange` event when this test runs. Doesn't happen in the browser. Unsure how to solve, tried to mock it with `e.stopPropagation` and `e.stopImmediatePropagation` but no joy..